Codec/Archive/Tar/Unpack.hs

```diff
@@ -122,3 +135,22 @@ setModTime path t =
 #else
 setModTime _path _t = return ()
 #endif
+
+setOwnerPermissions :: FilePath -> Permissions -> IO ()
+setOwnerPermissions path permissions =
+  setPermissions path ownerPermissions
+  where
+#if MIN_VERSION_directory(1,1,0)
+    ownerPermissions =
+      setOwnerReadable   (testBit permissions 8) $
```

Comment:
1. Let's float parsing `permissions` out of CPP conditions. 
2. It would be nice to quote a source of magic numbers in comments. Not sure which is the best one, maybe https://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html.